jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.py

```diff
@@ -132,31 +137,62 @@ def _subset_selected_changed(self, event={}):
         try:
             self._selected_subset = self._get_region_from_subset(subset_selected)
             self._selected_subset.meta['label'] = subset_selected
+
+            if isinstance(self._selected_subset, CirclePixelRegion):
+                self.bg_annulus_inner_r = self._selected_subset.radius
+            elif isinstance(self._selected_subset, (EllipsePixelRegion, RectanglePixelRegion)):
+                self.bg_annulus_inner_r = max(self._selected_subset.width,
+                                              self._selected_subset.height) * 0.5
+            else:  # pragma: no cover
+                raise TypeError(f'Unsupported region shape: {self._selected_subset.__class__}')
+
         except Exception as e:
             self._selected_subset = None
             self.reset_results()
             self.hub.broadcast(SnackbarMessage(
                 f"Failed to extract {subset_selected}: {repr(e)}", color='error', sender=self))
 
+    def _calc_bg_subset_median(self, reg):
+        # Basically same way image stats are calculated in vue_do_aper_phot()
+        # except here we only care about one stat for the background.
+        data = self._selected_data
+        comp = data.get_component(data.main_components[0])
+        aper_mask_stat = reg.to_mask(mode='center')
+        img_stat = aper_mask_stat.get_values(comp.data, mask=None)
+
+        # photutils/background/_utils.py --> nanmedian()
+        return np.nanmedian(img_stat)  # Naturally in data unit
+
+    @observe('bg_annulus_inner_r', 'bg_annulus_width')
+    def _bg_annulus_updated(self, *args):
+        if self.bg_subset_selected != 'Annulus':
+            return
+
+        try:
+            inner_r = float(self.bg_annulus_inner_r)
+            reg = CircleAnnulusPixelRegion(
+                self._selected_subset.center, inner_radius=inner_r,
+                outer_radius=inner_r + float(self.bg_annulus_width))
+            self.background_value = self._calc_bg_subset_median(reg)
+
+        except Exception as e:
+            self.background_value = 0
+            self.hub.broadcast(SnackbarMessage(
+                f"Failed to extract annulus: {repr(e)}", color='error', sender=self))

```

Comment:
It is a general catch-all. Like if someone tries to grab annulus with all NaNs in it or whatever crazy use case that I can not yet imagine.

> a very long queue of snackbar messages

But how? 😆

Reply:
I think I just changed the input multiple times that each raised an error here and they built up and took forever to clear because of the timeouts.

I think the catch-all here is probably fine, but any scenarios that we can predict, should avoid the snackbar if they are caught by validation rules (like a negative or 0 input for width).

Reply:
I'll just not emit any snackbar message here for now. In the future, if people want it back for certain things, we can add it back in; that is easier than guessing what they want. I am no Mel Gibson (obscure movie reference).

Reply:
I'm not sure this is better as now it just silently ignores and sets some values without any feedback.  Its very easy to get the UI in a state where you're not sure what is set automatically, what was set manually, and what was set but failed to propagate.  If you can put together a list of these scenarios that you expect to trip the backend, I can look into raising them in the input validation so the user gets feedback on the element itself and without the queued snackbars.

Reply:
> If you can put together a list of these scenarios

Can we defer this as a follow-up ticket? I don't think I can give you a thorough list in my rush today.

Reply:
Re: a list of these scenarios -- Okay, so coming back to this after a month... let's refresh our memories, this exception catching handles three different operations:

```python
inner_r = float(self.bg_annulus_inner_r)
reg = CircleAnnulusPixelRegion(
    self._selected_subset.center, inner_radius=inner_r,
    outer_radius=inner_r + float(self.bg_annulus_width))
self.background_value = self._calc_bg_subset_median(reg)
```

Let's break this down below.

1. `inner_r = float(self.bg_annulus_inner_r)`

This may break if `self.bg_annulus_inner_r` has string that cannot be converted to float. Maybe your JS-side check is preventing this happening from the GUI, but it is still possible from the Python side (e.g., from tests or direct call from other parts in Jdaviz), unless your template mixin thingy also prevents this?

2. `reg = CircleAnnulusPixelRegion(...)`

For this line, two things:

a. `float(self.bg_annulus_width)` can fail in the same way as `float(self.bg_annulus_inner_r)`.
b. Any checks that would fail in [CircleAnnulusPixelRegion constructor](https://github.com/astropy/regions/blob/9bd81c2d08eae589a3931f688298060d4505274d/regions/shapes/annulus.py#L83) would trigger a failure here. So, if `outer radius <= inner radius` (it is possible if you force width to be zero or negative) and less obviously, the class attribute checks that are baked into the inherited metaclasses.

3. `self.background_value = self._calc_bg_subset_median(reg)`

For this lines, several things:

a. It extracts the region mask from selected data. Region mask is defined w.r.t. *reference data*, which might not be the selected data. So if you loaded some crazy stuff, there is a chance this might fail.
b. Then it extracts the selected data values from the mask. If the above step succeeds, this might not pose any problem.
c. Finally, it calls `np.nanmedian`, which I think at most emits warning if it sees all NaNs or empty array. Not sure if we can easily crash this but I don't wanna jinx it.

### Now what?

With the current code at the time of posting this, I removed the snackbar message. So if any of the above results in error, you see background set to zero but no message on what went wrong.

And maybe this is okay, given error only happens if something is very wrong?

Reply:
I think this is a reasonable solution for now, thanks.  The most common case that you can easily trigger from the UI is a non-positive inner-radius or width.  These now raise the validation warning on the input to blame, display the background as zero, but allow the rest to be computed (all without queuing endless snackbars).  The other scenarios you brought up would likely require intentionally trying to break the plugin, and since the UI still provides feedback of what value is being adopted for the background (0), I don't think they're worth a ton of effort to manually catch now.